graph/patterns/navigation-property.md

```diff
@@ -0,0 +1,190 @@
+
+Navigation properties are also useful when clients sometimes want to retrieve both the primary entity and the related entity in a single round trip.  The `expand` query parameter makes this possible.
+
+### "parent-child" relationships
+
+Resources that contain a parent Id property in a child resource can utilize a navigation property in the parent resource that is declared as a collection of child resources. If desirable, a parent navigation property can also be created in the child resource to the parent resource. This is usually not necessary as the parent URL is a subset of child resource URL. The main use of this would be when retrieving child resources and choosing to expand properties of the parent resource so that both can be retrieved in a single request.  
+
+`/invoice/{invoiceId}/items/{itemId}?expand=parentInvoice(select=invoiceDate,Customer)`
+
+### "child-parent" relationships
+
+One other use case is when child resources appear in a non-contained collection and there is a desire to access the canonical parent:
+
+`/me/pinnedChannels/{channelId}/team`
+
+## Example
+-------
+
+### Retrieving a related entity
+
+```http
+GET /users/{id}/manager
+
+200 OK
+Content-Type: application/json
+
+{
+  "@odata.type": "#microsoft.graph.user",
+  "id": "6b3ee805-c449-46a8-aac8-8ff9cff5d213",
+  "displayName": "Bob Boyce"
+}
+```
+
+This navigation property could be described with the following CSDL: 
+```xml
+<EntityType Name="user">
+  <NavigationProperty Name="manager" Type="user" ContainsTarget="false" >
+</EntityType>
+```
+
+### Retrieving a reference to a related entity
+
+```http
+GET /users/{id}/manager/$ref
+
+200 OK
+Content-Type: application/json
+
+{
+  "@odata.id": "https://graph.microsoft.com/v1.0/directoryObjects/6b3ee805-c449-46a8-aac8-8ff9cff5d213/Microsoft.DirectoryServices.User"
+}
+```
+Note: Currently the base URL returned in $ref results are incorrect.  In order to process these URLs the client will need to convert the URL to a Graph URL.
+
+### Retrieving an entity with a related entity included
+
+```http
+GET /users/{id}?select=id,displayName&expand=manager(select=id,displayName)
+
+200 OK
+Content-Type: application/json
+
+{
+  "id": "3f057904-f936-4bf0-9fcc-c1e6f84289d8",
+  "displayName": "Jim James",
+  "manager": {
+    "@odata.type": "#microsoft.graph.user",
+    "id": "6b3ee805-c449-46a8-aac8-8ff9cff5d213",
+    "displayName": "Bob Boyce"
+  }
+}
+```
+
+### Creating an entity with a reference to a related entity
+
+Create a new user that references an existing manager
+```http
+POST /users/{id}
+Content-Type: application/json
+
+{
+    "displayName": "Bob",
+    "manager@bind": "https://graph.microsoft.com/v1.0/users/{managerId}"
+}
+
+201 Created
+```
+
+Create a new user and the users manager and create a relationship between the two.
+
+```http
+POST /users/{id}
+Content-Type: application/json
+
+{
+    "displayName": "Jim James",
+    "manager": {
+        "displayName": "Bob Boyce"
+    }
+}
+
```

Comment:
this example doesn't work on my test tenant. probably the URL should be POST /users?